src/metabase/models/collection.clj

```diff
@@ -1150,7 +1150,10 @@
   [collections]
   (if (= 1 (count collections))
     (let [collection (first collections)]
-      [(assoc collection :is_personal (is-personal-collection-or-descendant-of-one? collection))])
+      (if (some? collection)
+        [(assoc collection :is_personal (is-personal-collection-or-descendant-of-one? collection))]
+        ;; root collection is nil
+        [collection]))
```

Comment:
You should also handle the case where there is a root collection in `collections` and `collections` has more than one element. And add a test

Reply:
did that in https://github.com/metabase/metabase/pull/34827/commits/e31da20b1aee4fcd0ec93ca69774484032ca7d8b